lib/smart_answer/calculators/holiday_entitlement.rb

```diff
@@ -37,6 +41,48 @@ def initialize(working_days_per_week: 0,
       @days_per_shift_pattern = BigDecimal(days_per_shift_pattern, 10)
     end
 
+    def start_date=(date)
+      @start_date = date
+      @leave_year_start_date = calculate_leave_year_start_date
+    end
+
+    def leaving_date=(date)
+      @leaving_date = date
+      @leave_year_start_date = calculate_leave_year_start_date
```

Comment:
`calculate_leave_year_start_date` tests `worked_partial_year?` which tests the presence of both `start_date` and `leaving_date`.  Without running `calculate_leave_year_start_date` when either the `start_date` or `leaving_date` changes, we get a lot of failing tests.  When you dig into this it's because the `leave_year_start_date` is out.

Reply:
OK - It will have to stay as it is then :shrug: